config/plugins/interactive_environments/rstudio/templates/rstudio.mako

```diff
@@ -7,46 +8,42 @@ import time
 # Sets ID and sets up a lot of other variables
 ie_request.load_deploy_config()
 ie_request.attr.docker_port = 80
-# Create tempdir in galaxy
-temp_dir = ie_request.temp_dir
 PASSWORD = "rstudio"
 USERNAME = "rstudio"
 # Then override it again
 ie_request.notebook_pw = "rstudio"
 
-# Did the user give us an RData file?
-if hda.datatype.__class__.__name__ == "RData":
-    shutil.copy( hda.file_name, os.path.join(temp_dir, '.RData') )
+DATASET_HID = hda.hid
 
+# Add all environment variables collected from Galaxy's IE infrastructure
 ie_request.launch(
     image=trans.request.params.get('image_tag', None),
     additional_ids=trans.request.params.get('additional_dataset_ids', None),
     env_override={
         'notebook_username': USERNAME,
         'notebook_password': PASSWORD,
+        'dataset_hid': DATASET_HID,
```

Comment:
Ok, looks good otherwise.
The IE bug is probably due to that people never used it with multiple datasets.
Stil downloading the image ...